### PR TITLE
Make release workflow simpler and more automated

### PR DESCRIPTION
**Pull request by t1m0thyj**
## Proposed changes

This PR updates the Zowe Explorer deployment workflow to use [zowe-actions/octorelease](https://github.com/zowe-actions/octorelease).

Benefits of switching to Octorelease:
* It can automate commits that bump version and update changelog headers for us. For example:
  * Manual release PR: https://github.com/zowe/vscode-extension-for-zowe/pull/2089  vs  Automated release commit: https://github.com/t1m0thyj/vscode-extension-for-zowe/commit/363fc056a4268a7de0d5834f304b8e459f7dad57
  * Manual snapshot PR: https://github.com/zowe/vscode-extension-for-zowe/pull/2090  vs  Automated snapshot commit: https://github.com/t1m0thyj/vscode-extension-for-zowe/commit/64d6ef476b8eb2852b4fae74290e2821fb755acf
* The release process is implemented so that if any step fails, after fixing the issue you can just rerun the deployment workflow and any steps that succeeded previously will be skipped.
* The CLI squad already uses it to manage releases for all their repos, so less maintenance should be required if we share the same codebase 🙂 

TODO:
* [x] Merge https://github.com/zowe-actions/octorelease/pull/81
* [x] Replace the branch name `support-zowe-explorer` with `master` in workflow
* [ ] Update release process documentation in [the wiki](https://github.com/zowe/vscode-extension-for-zowe/wiki/Preparing-for-a-release)

## Release Notes

Milestone: 2.6.1

Changelog: N/A

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ ] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Review by zFernand0 — COMMENTED** [review]
Overall, the changes seem to work great.
Thanks for providing such automation.
Hopefully our release process gets a lot smoother 😋 

P.S.: will approve once the other PR has been merged and `support-zowe-explorer -> master` (or v1 😉)

**Review by JillieBeanSim — APPROVED** [review]

**Review by zFernand0 — DISMISSED** [review]

**Review by traeok — APPROVED** [review]

**Review by zFernand0 — APPROVED** [review]

**Comment by zFernand0** [comment]
Mac runners seem to be struggling.
we can merge this PR even if any of them timeout.

**Pull request merged**
